libraries/AP_Common/AP_BackendIterator.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <stdint.h>
+
+template <typename T>
+class AP_BackendIterator
+{
+public:
+    AP_BackendIterator(T**_backends, uint8_t &_num_backends) :
+        backends{_backends},
+        num_backends{_num_backends}
+        {}
+
+    // see
+    // https://www.internalpointers.com/post/writing-custom-iterators-modern-cpp
+    // - search for "done by implementing some custom"
+    struct Iterator {
+        Iterator(T **_backend) : backend{_backend} {}
+        T& operator*() const { return **backend; }
+        T* operator->() { return *backend; }
+
+        // Prefix increment
+        Iterator& operator++() { backend++; return *this; };
+        // Postfix increment
+        Iterator operator++(int) { Iterator tmp = *this; ++(*this); return tmp; }
```

Comment:
do we need to use "int" for some reason?  or maybe it should be uint8_t?

Reply:
This is actually just to disambiguate prefix vs postfix in the C++ signature for the operators for pre-and-post-fix.  That is, they didn't do this:
++operator()
operator++()
 - they added the `int` in there to make that one the postfix operator.
 

So I think it has to stay as `int` because it's basically C++ needing to disamabiguate.